### ISSUE TITLE
Desktop client crashes on trying to re-enter a group call

### ISSUE THREAD

**Issue by binford6000** [nextcloud/talk-desktop#838]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Start a group call
2. Change to any 1:1 chat room with CTRL-/⌘-Click
3. Try to change back to group call room by clicking on the room

### Expected behaviour
"Leave this page" should return to the 1:1 window and (maybe) "join here shouldn't be available at all:
![Image](https://github.com/user-attachments/assets/fd0a648e-c059-4ab1-bf5c-2ba38ed596e1)

### Actual behaviour

- On windows 11 systems the 1:1-chat window crashes the client. It needs to be restarted again.

- On macOS (10.15.) the 1:1 window crashes (you can close it) and in the back you drop out of the group call:

![Image](https://github.com/user-attachments/assets/baf59971-1e43-4b5f-9163-f0b50d275916)

![Image](https://github.com/user-attachments/assets/62041041-c6f1-45d8-9780-565ee3b57851)

## Desktop client

**Talk Desktop client version:** v0.30.0

**Operating system:** Windows 11 / macOS 10.15

**Operating system version:** Windows 11 / (In case of Linux, provide the distribution.)

**Microphone available:** yes/no

**Camera available:** yes/no

## Server

**Nextcloud version: 30.0.0** 

**Talk app version: 20.0.0** 

**Custom Signaling server configured:** yes (2.0.0) 

**Custom TURN server configured:** yes 

**Custom STUN server configured:** yes

## Logs

No logs available.

### Client log

----------------------------System report----------------------------
Nextcloud Talk Desktop version 0.39.0
- Built with Nextcloud Talk version 20.0.1

Connected to:
- Server address: https://try.nextcloud.com/ltd/sebastian
- Nextcloud Server version 30.0.0
- Nextcloud Talk version 20.0.0

OS: Darwin 24.0.0 (Darwin Kernel Version 24.0.0: Tue Sep 24 23:36:26 PDT 2024; root:xnu-11215.1.12~1/RELEASE_ARM64_T8103)
----------------------------System report----------------------------
<details>

```
Insert your browser log here, this could for example include:
a) The javascript console log
b) The network log
c) ...
```

</details>

**Comment by nickvergessen** [comment]
Technical background: you cannot join the same conversation twice from the same browser/desktop client.
Maybe the desktop client could handle it in a way that it knows that you have that call open in another window and simply bring that to the foreground?

**Comment by ShGKme** [comment]
> 2\. Change to any 1:1 chat room with CTRL-/⌘-Click

This feature does not officially exist 👀

**Comment by binford6000** [comment]
> Maybe the desktop client could handle it in a way that it knows that you have that call open in another window and simply bring that to the foreground?

Yes, that would be an option as well. Or as mentioned above an additional message that says that you can't join twice not showing the 'join' button at all?

If you do nothing at all in the web UI you are redirected to 
`https://example.com/index.php/apps/spreed/duplicate-session` 

![Image](https://github.com/user-attachments/assets/96e5a4eb-dadb-45d2-b8b2-3a430819cd4d)

Maybe another option would be to get back to the talk room you tried to join the duplicate call from?

**Comment by binford6000** [comment]
> This feature does not officially exist 👀

I took over that POC and I *think* they had the hint from us... 😬
